### PR TITLE
Improve dot positioning in multiple voices

### PR DESCRIPTION
**Pull request by Monceber**
Changed logic of dot placement so that they are shifted up/down based on the voice that they are in (shifted down for lower voices and vice versa) so that there is no confusion where note belongs to:

**Before:**
image
**After:**
image

<details><summary>Example file:</summary>

```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
   <meiHead>
      <fileDesc>
         <titleStmt>
            <title />
            <respStmt />
         </titleStmt>
         <pubStmt><date isodate="2021-03-03" type="encoding-date">2021-03-03</date>
         </pubStmt>
      </fileDesc>
      <encodingDesc xml:id="encodingdesc-0000000000002481">
         <appInfo xml:id="appinfo-0000000000018598">
            <application xml:id="application-0000000000021576" isodate="2021-03-31T11:51:41" version="3.4.0-dev-[undefined]">
               <name xml:id="name-0000000000016523">Verovio</name>
               <p xml:id="p-0000000000021560">Transcoded from MusicXML</p>
            </application>
         </appInfo>
      </encodingDesc>
   </meiHead>
   <music>
      <body>
         <mdiv xml:id="mdiv-0000000000015873">
            <score xml:id="score-0000000000018344">
               <scoreDef xml:id="scoredef-0000000000009784">
                  <staffGrp xml:id="staffgrp-0000000000008067">
                     <staffDef xml:id="P1" n="1" lines="5" ppq="4">
                        <label xml:id="label-0000000000008960">Piano</label>
                        <labelAbbr xml:id="labelAbbr-0000000000018810">Pno.</labelAbbr>
                        <instrDef xml:id="instrdef-0000000000008284" midi.channel="0" midi.instrnum="0" midi.volume="78.00%" />
                        <clef xml:id="clef-0000000000002343" shape="G" line="2" />
                        <keySig xml:id="keysig-0000000000027960" sig="0" />
                        <meterSig xml:id="msig-0000000000031914" count="4" unit="4" />
                     </staffDef>
                  </staffGrp>
               </scoreDef>
               <section xml:id="section-0000000000030039">
                  <pb xml:id="pb-0000000000027985" />
                  <measure xml:id="measure-0000000000018512" n="1">
                     <staff xml:id="staff-0000000000020366" n="1">
                        <layer xml:id="layer-0000000000010775" n="1">
                           <note xml:id="note-0000000000014732" dots="1" dur.ppq="6" dur="4" oct="5" pname="d" stem.dir="up" />
                           <note xml:id="note-0000000000019441" dots="1" dur.ppq="6" dur="4" oct="4" pname="b" stem.dir="up" />
                           <note xml:id="note-0000000000014502" dur.ppq="4" dur="4" oct="5" pname="d" stem.dir="up" />
                        </layer>
                        <layer xml:id="layer-0000000000005193" n="2">
                           <note xml:id="note-0000000000013259" dots="1" dur.ppq="6" dur="4" oct="4" pname="e" stem.dir="down" />
                           <note xml:id="note-0000000000014070" dots="1" dur.ppq="6" dur="4" oct="4" pname="g" stem.dir="down" />
                           <beam xml:id="beam-0000000000023947">
                              <note xml:id="note-0000000000006653" dots="1" dur.ppq="3" dur="8" oct="4" pname="b" stem.dir="down" />
                              <note xml:id="note-0000000000021191" dur.ppq="1" dur="16" oct="4" pname="a" stem.dir="down" />
                           </beam>
                        </layer>
                     </staff>
                  </measure>
                  <scoreDef xml:id="scoredef-0000000000014363">
                     <meterSig xml:id="msig-0000000000027895" count="6" unit="4" />
                  </scoreDef>
                  <measure xml:id="measure-0000000000008697" n="2">
                     <staff xml:id="staff-0000000000010773" n="1">
                        <layer xml:id="layer-0000000000030349" n="1">
                           <note xml:id="note-0000000000021115" dots="1" dur.ppq="12" dur="2" oct="5" pname="c" stem.dir="up" />
                           <note xml:id="note-0000000000016657" dots="1" dur.ppq="12" dur="2" oct="4" pname="b" stem.dir="up" />
                        </layer>
                        <layer xml:id="layer-0000000000026848" n="2">
                           <note xml:id="note-0000000000026283" dots="1" dur.ppq="12" dur="2" oct="4" pname="b" stem.dir="down" />
                           <note xml:id="note-0000000000024023" dots="1" dur.ppq="12" dur="2" oct="4" pname="a" stem.dir="down" />
                        </layer>
                     </staff>
                  </measure>
                  <scoreDef xml:id="scoredef-0000000000009630">
                     <meterSig xml:id="msig-0000000000006094" count="6" unit="8" />
                  </scoreDef>
                  <measure xml:id="measure-0000000000030236" n="3">
                     <staff xml:id="staff-0000000000001194" n="1">
                        <layer xml:id="layer-0000000000023179" n="1">
                           <note xml:id="note-0000000000015642" dots="1" dur.ppq="6" dur="4" oct="5" pname="c" stem.dir="up" />
                           <note xml:id="note-0000000000028939" dots="1" dur.ppq="6" dur="4" oct="4" pname="b" stem.dir="up" />
                        </layer>
                        <layer xml:id="layer-0000000000028304" n="2">
                           <note xml:id="note-0000000000025959" dots="1" dur.ppq="6" dur="4" oct="4" pname="b" stem.dir="down" />
                           <note xml:id="note-0000000000016475" dots="1" dur.ppq="6" dur="4" oct="4" pname="a" stem.dir="down" />
                        </layer>
                     </staff>
                  </measure>
                  <scoreDef xml:id="scoredef-0000000000007290">
                     <meterSig xml:id="msig-0000000000012609" count="3" unit="4" />
                  </scoreDef>
                  <measure xml:id="measure-0000000000002054" n="4">
                     <staff xml:id="staff-0000000000025130" n="1">
                        <layer xml:id="layer-0000000000004017" n="1">
                           <note xml:id="note-0000000000009797" dots="1" dur.ppq="12" dur="2" oct="5" pname="c" stem.dir="up" />
                        </layer>
                        <layer xml:id="layer-0000000000029774" n="2">
                           <note xml:id="note-0000000000007823" dur.ppq="8" dur="2" oct="4" pname="b" stem.dir="down" />
                           <note xml:id="note-0000000000017833" dur.ppq="4" dur="4" oct="4" pname="a" stem.dir="down" />
                        </layer>
                     </staff>
                  </measure>
                  <measure xml:id="measure-0000000000030200" right="end" n="5">
                     <staff xml:id="staff-0000000000018481" n="1">
                        <layer xml:id="layer-0000000000001380" n="1">
                           <note xml:id="note-0000000000030327" dots="1" dur.ppq="12" dur="2" oct="4" pname="b" stem.dir="up" />
                        </layer>
                        <layer xml:id="layer-0000000000003436" n="2">
                           <note xml:id="note-0000000000007539" dur.ppq="8" dur="2" oct="4" pname="a" stem.dir="down" />
                           <note xml:id="note-0000000000029788" dur.ppq="4" dur="4" oct="4" pname="g" stem.dir="down" />
                        </layer>
                     </staff>
                  </measure>
               </section>
            </score>
         </mdiv>
      </body>
   </music>
</mei>

```

</details>

### CODE REVIEW

**Comment by ahankinson** [comment]
Can you give some engraved examples or a reference in Gould for this? Putting the dot below the line looks very strange.

**Comment by rettinghaus** [comment]
@ahankinson Here's what Gould tells us:
image-20210303-101915

File: src/chord.cpp [hunk 1]
```diff
@@ -673,65 +742,25 @@ int Chord::CalcDots(FunctorParams *functorParams)
...
-            if (loc < this->GetBottomNote()->GetDrawingLoc() - 1) continue;
+        if (noteLocations.end() == noteLocations.find(staff)) {
+            noteLocations[staff] = {};
         }
+        noteLocations[staff].insert(note->GetDrawingLoc());
+    }
 
-        // if it's not, add it to the dots list and go back to DrawChord
-        dotLocs->push_back(loc);
+    for (auto loc : noteLocations) {
```
Comment (lpugin) [inline src/chord.cpp:762]: Wouldn't it be better to use a reference `auto &loc` here?

Comment (Monceber) [reply src/chord.cpp:762]: Yes, you're right, thanks!

**Comment by lpugin** [comment]
@rettinghaus could you add the example under `dot` in the test suite? Thanks

**Pull request merged**